### DISCUSSION TITLE
Twoslash integration

### DISCUSSION THREAD

**Discussion by o-az** [withastro/starlight#1381] [Feature Requests]
### What version of `starlight` are you using?

latest

### What is your idea?

This would be an amazing addition to starlight 

https://github.com/antfu/shikiji/tree/main/packages/shikiji-twoslash

### Why is this feature necessary?

Get more insight into code blocks

### Do you have examples of this feature in other projects?

https://shikiji.netlify.app/packages/vitepress#twoslash

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by HiDeoo** [comment]
Thanks for the feedback 🙌 

For reference, Starlight’s code blocks are using Expressive Code which is currently tracking this in https://github.com/expressive-code/expressive-code/issues/95

**Reply by delucis** [reply]
+1 — Starlight would likely rely on Expressive Code for this, and I think they will likely help improve some of the a11y & UX issues present in current implementations of this kind of thing.

**Comment by o-az** [comment]
Great to hear! That should do it. Should I close this? @HiDeoo

**Reply by HiDeoo** [reply]
Considering this is a discussion, I think it's fine to keep it opened, I was mostly linking it for reference purposes. As this is also a topic having a lot of activity these days, the discussion could also be useful to discuss some details considering the implementation may differ with the linked examples to avoid some accessibility issues.

**Comment by HiDeoo** [comment]
Small update: there is now a [`expressive-code-twoslash` plugin](https://twoslash.matthiesen.dev/) available for Expressive Code.

Also worth linking the Expressive Code community plugins list: https://expressive-code.com/plugins/community-plugins/
